components/encoder/incremental/incremental_test.go

```diff
@@ -160,26 +165,59 @@ func TestEncoder(t *testing.T) {
 	})
 }
 
-func MakeBoard(t *testing.T) *fakeboard.Board {
-	interrupt11, _ := fakeboard.NewDigitalInterruptWrapper(board.DigitalInterruptConfig{
-		Name: "11",
-		Pin:  "11",
-	})
-
-	interrupt13, _ := fakeboard.NewDigitalInterruptWrapper(board.DigitalInterruptConfig{
-		Name: "13",
-		Pin:  "13",
-	})
+func MakeBoard(t *testing.T) board.Board {
+	b := inject.NewBoard("test-board")
+	i1 := &inject.DigitalInterrupt{}
+	i2 := &inject.DigitalInterrupt{}
+	callbacks := make(map[board.DigitalInterrupt]chan board.Tick)
+	i1.NameFunc = func() string {
+		return "11"
+	}
+	i2.NameFunc = func() string {
+		return "13"
+	}
+	i1.TickFunc = func(ctx context.Context, high bool, nanoseconds uint64) error {
+		ch, ok := callbacks[i1]
+		test.That(t, ok, test.ShouldBeTrue)
+		ch <- board.Tick{Name: i1.Name(), High: high, TimestampNanosec: nanoseconds}
+		return nil
+	}
+	i2.TickFunc = func(ctx context.Context, high bool, nanoseconds uint64) error {
+		ch, ok := callbacks[i2]
+		test.That(t, ok, test.ShouldBeTrue)
+		ch <- board.Tick{Name: i2.Name(), High: high, TimestampNanosec: nanoseconds}
+		return nil
+	}
+	i1.ValueFunc = func(ctx context.Context, extra map[string]interface{}) (int64, error) {
+		return 0, nil
+	}
+	i2.ValueFunc = func(ctx context.Context, extra map[string]interface{}) (int64, error) {
+		return 0, nil
+	}
+	i1.RemoveCallbackFunc = func(c chan board.Tick) {
+		delete(callbacks, i1)
+	}
+	i2.RemoveCallbackFunc = func(c chan board.Tick) {
+		delete(callbacks, i2)
+	}
 
-	interrupts := map[string]*fakeboard.DigitalInterruptWrapper{
-		"11": interrupt11,
-		"13": interrupt13,
+	b.DigitalInterruptByNameFunc = func(name string) (board.DigitalInterrupt, bool) {
```

Comment:
same here this needs to be returning an error, might want to merge main into your branch to make sure all these cases are caught